### PR TITLE
Changes to allow add lib to cmake projects as subdirectory

### PR DESCRIPTION
**Pull request by ayfomin**
This simplifies use lib in cmake projects. It could be added to project as git submodule and included in CMakeLists.txt as follow: 
`add_subdirectory(path_to_quickfix)`
`target_link_libraries(target_name PRIVATE quickfix)` 
There is no need to build and install library separatly. It will be built within your project.

### CODE REVIEW

**Comment by karopawil** [comment]
Looks good to me.

File: src/C++/CMakeLists.txt [hunk 1]
```diff
@@ -71,12 +71,27 @@ else()
   target_link_libraries(${PROJECT_NAME} ${OPENSSL_LIBRARIES} ${MYSQL_CLIENT_LIBS} ${PostgreSQL_LIBRARIES} pthread)
 endif()
 
-target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src/C++)
+file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/ DESTINATION include/quickfix
```
Comment (joaosoares) [inline src/C++/CMakeLists.txt]: I'm getting a recursion error on this line for the file `OrderCancelReplaceRequest.h`, after including it in my project as mentioned in the PR description. Any reasons why?

```
CMake Error at vendor/quickfix/src/C++/CMakeLists.txt:74 (file):
  file COPY cannot copy file
  "/home/jsoares/Projects/atlas/fix_listener_cpp/vendor/quickfix/src/C++/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/fix43/OrderCancelReplaceRequest.h"
  to
  "/home/jsoares/Projects/atlas/fix_listener_cpp/vendor/quickfix/src/C++/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/include/quickfix/fix43/OrderCancelReplaceRequest.h".
```

Comment (ayfomin) [reply src/C++/CMakeLists.txt]: It could happen if you use 'in-source' build. I'll fix it.

Comment (joaosoares) [reply src/C++/CMakeLists.txt]: It works now. Thanks!

**Comment by bajajmanoj** [comment]
Why aren't we using [PROJECT_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/PROJECT_SOURCE_DIR.html), rather than using CMAKE_CURRENT_SOURCE_DIR

**Comment by orenmnero** [comment]
Fixed in other PRs

**Pull request closed without merging**